editor/components/block-switcher/test/index.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+//import React from 'react';
+import renderer from 'react-test-renderer';
+
+/**
+ * Internal dependencies
+ */
+import { BlockSwitcher } from '../';
+
+describe( 'BlockSwitcher', () => {
+	test( 'Test block switcher without blocks', () => {
+		const tree = renderer
+			.create( <BlockSwitcher /> )
+			.toJSON();
+
+		expect( tree ).toMatchSnapshot();
+	} );
+	test( 'Test block switcher with blocks', () => {
+		const block = {
+			attributes: {
+				content: [ 'How are you?' ],
+				nodeName: 'H2',
+			},
+			isValid: true,
+			name: 'core/heading',
+			originalContent: '<h2>How are you?</h2>',
+			uid: 'a1303fd6-3e60-4fff-a770-0e0ea656c5b9',
+		};
+
+		const blocks = [
+			block,
+		];
+
+		const tree = renderer
```

Comment:
This way is promoted by Jest team in their official docs. I personally prefer using `enzyme` to test React components, but this it might be an easier way in case when you have only snapshot tests in the file. Enzyme is more helpful when doing more complex checks.

Reply:
Yeah, Enzyme is great for getting into the details. Not sure, how we want to use snapshots etc.